src/main/java/org/broadinstitute/hellbender/engine/filters/ReadFilterLibrary.java

```diff
@@ -109,42 +124,53 @@ private ReadFilterLibrary(){ /*no instance*/ }
                     .anyMatch(c -> c.getOperator().consumesReferenceBases() && c.getLength() > 0);
         }}
 
+    /** Filter out reads marked as duplicates. */
     @DocumentedFeature(groupName=HelpConstants.DOC_CAT_READFILTERS, groupSummary=HelpConstants.DOC_CAT_READFILTERS_SUMMARY)
     public static class NotDuplicateReadFilter extends ReadFilter {
         private static final long serialVersionUID = 1L;
         @Override public boolean test(final GATKRead read){
             return ! read.isDuplicate();}}
 
+    /** Filter out reads representing secondary alignments. */
     @DocumentedFeature(groupName=HelpConstants.DOC_CAT_READFILTERS, groupSummary=HelpConstants.DOC_CAT_READFILTERS_SUMMARY)
     public static class NotSecondaryAlignmentReadFilter extends ReadFilter {
         private static final long serialVersionUID = 1L;
         @Override public boolean test(final GATKRead read) {
             return !read.isSecondaryAlignment();}}
 
+    /** Filter out reads representing supplementary alignments. */
     @DocumentedFeature(groupName=HelpConstants.DOC_CAT_READFILTERS, groupSummary=HelpConstants.DOC_CAT_READFILTERS_SUMMARY)
     public static class NotSupplementaryAlignmentReadFilter extends ReadFilter {
         private static final long serialVersionUID = 1L;
         @Override public boolean test(final GATKRead read) {
             return !read.isSupplementaryAlignment();}}
 
+    /** Filter out unpaired reads. */
     @DocumentedFeature(groupName=HelpConstants.DOC_CAT_READFILTERS, groupSummary=HelpConstants.DOC_CAT_READFILTERS_SUMMARY)
     public static class PairedReadFilter extends ReadFilter {
         private static final long serialVersionUID = 1L;
         @Override public boolean test(final GATKRead read) {
             return read.isPaired();}}
 
+    /** Filter out reads failing platform/vendor quality checks. */
     @DocumentedFeature(groupName=HelpConstants.DOC_CAT_READFILTERS, groupSummary=HelpConstants.DOC_CAT_READFILTERS_SUMMARY)
     public static class PassesVendorQualityCheckReadFilter extends ReadFilter {
         private static final long serialVersionUID = 1L;
         @Override public boolean test(final GATKRead read){
             return ! read.failsVendorQualityCheck();}}
 
+    /** Keep only reads that are properly paired. */
     @DocumentedFeature(groupName=HelpConstants.DOC_CAT_READFILTERS, groupSummary=HelpConstants.DOC_CAT_READFILTERS_SUMMARY)
     public static class ProperlyPairedReadFilter extends ReadFilter {
         private static final long serialVersionUID = 1L;
         @Override public boolean test(final GATKRead read) {
             return read.isProperlyPaired();}}
 
+    /**
+     * Keep reads representing primary alignments.
```

Comment:
Okay. So you are saying that either this filter will be renamed or will be fixed to actually do what its name implies. If these changes are certain to come in the near future, then perhaps we can warn folks ahead of time to anticipate changes. What do you think @vdauwera? After the warning, state that 

> The NotSecondaryAlignmentReadFilter also filters based solely on secondary alignment status. The PrimaryAlignmentReadFilter will therefore undergo changes in the future to also filter supplementary alignments.